### PR TITLE
translations: document weblate workflow

### PR DESCRIPTION
**Pull request by iGormilhit**

### CODE REVIEW

File: translations/translations-workflow.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Translations workflow
+
+We're using weblate as a web service to help the translation process, through
+the hosted service of weblate: https://hosted.weblate.org/projects/rero_plus
+
+For our needs, on weblate, we have one project (name RERO +, with `rero_plus`
```
Comment (blankoworld) [inline translations/translations-workflow.md]: I would prefer `named RERO +`, what do you think?

File: translations/translations-workflow.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- `ng-core`.
+- `rero-ils`.
+- `rero-ils-public-search` (from the GitHub `rero-ils-ui`).
+- `rero-ils-amdin` (from the GitHub `rero-ils-ui`).
+- `sonar`.
+- `sonar-ui`.
+
+## The *very* first time
+
+When every existing PR that contain modifications on the translated files are
```
Comment (blankoworld) [inline translations/translations-workflow.md]: `every existing PR` `contains` (with `s` I expected)

File: translations/translations-workflow.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  documentation][1].
+
+This ensure two things:
+
+1. Web changes are merged into the `dev` branch of our repository, the weblate
+   service is informed through the webhook and pulls changes to the weblate
+   `git` repository. It's done with a `git rebase` command, as it can be
+   configured in the component settings [component/settings/version control].
+1. As translations are being made through the weblate interface, commits are
+   being made. 24 hours after a commit, a PR is created on the corresponding
```
Comment (blankoworld) [inline translations/translations-workflow.md]: I would add a precision on the fact that 24 hours is configurable in Weblate administration panel (or somewhere else)

File: translations/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Translations
+
+- [How to add a language][1]
+- [Translations Workflow][2]
```
Comment (sebdeleze) [inline translations/README.md]: Maybe `worklow` lower case?

File: translations/translations-workflow.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+We're using weblate as a web service to help the translation process, through
+the hosted service of weblate: https://hosted.weblate.org/projects/rero_plus
+
+For our needs, on weblate, we have one project (named RERO +, with `rero_plus`
+as slug), with many *components*:
+
+- `ng-core`.
+- `rero-ils`.
+- `rero-ils-public-search` (from the GitHub `rero-ils-ui`).
+- `rero-ils-amdin` (from the GitHub `rero-ils-ui`).
```
Comment (sebdeleze) [inline translations/translations-workflow.md]: typo: `rero-ils-admin`

File: translations/translations-workflow.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+1. Verify that the weblate component is up to date with the GitHub project. To
+   do that you can either:
+    - Got to https://hosted.weblate.org/projects/rero_plus/ng-core/#repository
+      (change `ng-core` to your needs) and check the repository status.
+    - Or clone the weblate `git` repository and inspect the `git` logs (`git
+      clone
+      'https://hosted.weblate.org/git/rero_plus/<component>/')`.
+    If it not, check [the workflow][3] below, to know how to update it.
+2. Verify that on the GitHub repository settings, the webhook is active.
+3. Unlock the tranlsations (you can do that on the web interface, or through
```
Comment (sebdeleze) [inline translations/translations-workflow.md]: typo: `translations`

File: translations/translations-workflow.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+   the [`wlc` client][4]).
+4. Inform the translators that weblate is ready.
+
+## The regular workflow
+
+For the sychronisation workflow, it's semi automatic:
```
Comment (sebdeleze) [inline translations/translations-workflow.md]: `synchronisation`

File: translations/translations-workflow.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+ ## Some weblate settings
+
+- [component/settings/version control] Merge Style: rebase.
+- [component/settings/version control] Age of changes to commit: 24 [hour].
+- [rero_/#workflow] Source language: English (United States). This sets `en_US`
+  as the source language file, in order to avoid translating it.
+- [component/seggings/files]: filter: `^(?!(en_US)$).+$` (for `ng-core`,
+  `public-search`, `admin`). This is needed, as in our angular project, the
+  source language file is the same as a translated language files.
+- [component/seggings/files]: filter: `^(?!(messages)$).+$` (for `sonar-ui`).
```
Comment (sebdeleze) [inline translations/translations-workflow.md]: [seggings](https://www.urbandictionary.com/define.php?term=segging) ? I learned something today 😆

**Review by sebdeleze — CHANGES_REQUESTED** [review]

File: translations/translations-workflow.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+   service is informed through the webhook and pulls changes to the weblate
+   `git` repository. It's done with a `git rebase` command, as it can be
+   configured in the component settings [component/settings/version control].
+1. As translations are being made through the weblate interface, commits are
+   being made. 24 hours after a commit ([see settings section][5]), a PR is
+   created on the corresponding GitHub repository (called *upstream* by the
+   weblate documentation).
+
+As it is managed with `git`, and as we use *bilingual* files (each translation
+files has two languages), which suppose to update the catalogs, conflicts can
```
Comment (blankoworld) [inline translations/translations-workflow.md]: `supposes`?

File: translations/translations-workflow.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+happen. To avoid such conflicts, [the documentation][2] propose something like
```
Comment (blankoworld) [inline translations/translations-workflow.md]: `proposes`?

File: translations/translations-workflow.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+the following:
+
+*You'll need to [install and configure the `wlc` client][4]*.
+
+1. The `dev` branch (*upstream*) is moving. Developers do not
+   `extract_messages` nor `update_catalog`.
+1. From time to time, the person responsible for the translations update
```
Comment (blankoworld) [inline translations/translations-workflow.md]: `updates`?

**Review by blankoworld — APPROVED** [review]

**Review by sebdeleze — APPROVED** [review]

**Pull request merged**